src/test/mod.rs

```diff
@@ -49,11 +50,11 @@ use std::{fs::read_to_string, str::FromStr};
 const MAX_POOL_SIZE: u32 = 100;
 
 lazy_static! {
-    pub(crate) static ref CLIENT_OPTIONS: ClientOptions = {
-        let mut options = ClientOptions::parse_without_srv_resolution(&DEFAULT_URI).unwrap();
+    pub(crate) static ref CLIENT_OPTIONS: AsyncOnce<ClientOptions> = AsyncOnce::new(async {
```

Comment:
Nice writeup! One thing that makes this a bit more complicated is that some of the underlying implementations for the `runtime` functions we define don't exactly line up with the name we use for them. For instance, our `block_on` calls into `block_in_place` rather than `block_on` when using async tokio, which is why there's that multithreaded runtime requirement. We have RUST-800 as a quick win to fix this which will hopefully make these things easier to reason about.